deepcell_toolbox/utils.py

```diff
@@ -309,12 +311,28 @@ def resize(data, shape, data_format='channels_last'):
         else:
             shape = tuple(list(shape) + [data.shape[channel_axis]])
 
-        _resize = lambda d: transform.resize(d, shape, mode='constant', preserve_range=True)
+        # linear interpolation (order 1) for image data, nearest neighbor (order 0) for labels
+        # anti_aliasing introduces spurious labels, include only for image data
+        order = 0 if labeled_image else 1
+        anti_aliasing = not labeled_image
+
+        _resize = lambda d: transform.resize(d, shape, mode='constant', preserve_range=True,
+                                             order=order, anti_aliasing=anti_aliasing)
     # single channel image, resize with cv2
     else:
         shape = tuple(shape)
 
-        _resize = lambda d: np.expand_dims(cv2.resize(np.squeeze(d), shape), axis=channel_axis)
+        # linear interpolation for image data, nearest neighbor for labels
+        # CV2 doesn't support ints for linear interpolation, set to float for image data
+        if labeled_image:
+            interpolation = cv2.INTER_NEAREST
+        else:
+            interpolation = cv2.INTER_LINEAR
+            data = data.astype('float32')
```

Comment:
@msschwartz21, I moved the dtype conversion to happen here, since we only need to convert if using cv2, and only if using interpolation.

Do we need to convert back? Or can I remove that section from applications completely (https://github.com/vanvalenlab/deepcell-tf/blob/c9e174542dc4453049c4e4e1ce90f6738c6723a0/deepcell/applications/application.py#L245)

Reply:
My instinct is that we should preserve the dtype that we get as input and therefore convert it back to the original type. @willgraf do you have a preference?

Reply:
If I were using a resize function I would expect my resized data to be the same datatype as what I put in.

Honestly, I'm not sure I see a downside of using something like `return resized.astype(original_dtype)` to always recast even if its unnecessary.

Reply:
Agreed, just added that in